File: python/readability.py
def main():
    text = input("Text: ")

    # After every word we have a space, so the number of spaces = the number of words - 1
    words = text.count(" ") + 1

    # Counts all letters from the text and get the average from the words
    L = len([c for c in text if c.isalpha()]) * 100 / (words)

    # Every sentence ends in . ! ?
    S = (text.count(".") + text.count("?") + text.count("!")) * 100 / (words)

    # Uses the formula
    index = (0.0588 * L) - (0.296 * S) - 15.8

    # Prints the result
    if index >= 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {round(index)}")

if __name__ == '__main__':
    main()

File: arrays/readability.c
#include <stdio.h>
#include <string.h>
#include <cs50.h>
#include <math.h>
#include <ctype.h>

int main(void)
{
    // Getting the user's prompt
    string text = get_string("Text: ");

    // Setting up the variables
    int letters = 0;
    int words = 0;
    int sentences = 0;

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        // Checks if it's a letter, comparing with chars as ascii integers
        if ((text[i] >= 'A' && text[i] <= 'Z') || (text[i] >= 'a' && text[i] <= 'z'))
        {
            letters++;
        }
        // A space means a new word appeared
        else if (text[i] == ' ')
        {
            words++;
        }
        // A period, exclamation point or question mark means a new sentence is taking place
        else if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentences++;
        }
    }
    // We have to add one more to words because there's always one less space than words in sentences
    words++;

    // Calculate the averages
    float L = (float) letters / (float) words * 100;
    float S = (float) sentences / (float) words * 100;

    // Using the formula
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    // Adapting the output
    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }

}